Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/services/dbus.te
+++ b/policy/modules/services/dbus.te
@@ -104,7 +104,7 @@
 allow system_dbusd_t self:capability { dac_override kill setgid setpcap setuid sys_resource };
 # net_admin for changing buffer sizes
 dontaudit system_dbusd_t self:capability { net_admin sys_tty_config };
-allow system_dbusd_t self:process { getattr getcap getsched setcap setpgid setrlimit signal_perms };
+allow system_dbusd_t self:process { getattr getcap getsched setcap setpgid setrlimit signal_perms setsched };
 allow system_dbusd_t self:fifo_file rw_fifo_file_perms;
 allow system_dbusd_t self:dbus { acquire_svc send_msg };
 allow system_dbusd_t self:unix_stream_socket { connectto create_stream_socket_perms };
@@ -133,6 +133,7 @@
 
 kernel_read_system_state(system_dbusd_t)
 kernel_read_kernel_sysctls(system_dbusd_t)
+kernel_read_vm_overcommit_sysctl(system_dbusd_t)
 
 corecmd_list_bin(system_dbusd_t)
 corecmd_read_bin_pipes(system_dbusd_t)
