Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/system/sysnetwork.fc
+++ b/policy/modules/system/sysnetwork.fc
@@ -18,12 +18,14 @@
 /etc/hosts		--	gen_context(system_u:object_r:net_conf_t,s0)
 /etc/hosts\.allow.*	--	gen_context(system_u:object_r:net_conf_t,s0)
 /etc/hosts\.deny.*	--	gen_context(system_u:object_r:net_conf_t,s0)
+/etc/hosts		-l	gen_context(system_u:object_r:net_conf_t,s0)
+/run/hosts		--	gen_context(system_u:object_r:net_conf_t,s0)
 /etc/hostname		--	gen_context(system_u:object_r:net_conf_t,s0)
-/etc/kea(/.*)?			gen_context(system_u:object_r:dhcp_etc_t,s0)
 /etc/denyhosts.*	--	gen_context(system_u:object_r:net_conf_t,s0)
-/etc/machine-info	--	gen_context(system_u:object_r:net_conf_t,s0)
-/etc/resolv\.conf.*	--	gen_context(system_u:object_r:net_conf_t,s0)
+/etc/resolv\.conf.*		gen_context(system_u:object_r:net_conf_t,s0)
 /etc/yp\.conf.*		--	gen_context(system_u:object_r:net_conf_t,s0)
+
+/etc/dhcp3(/.*)?		gen_context(system_u:object_r:dhcp_etc_t,s0)
 
 /etc/dhcp3(/.*)?		gen_context(system_u:object_r:dhcp_etc_t,s0)
 /etc/dhcp3?/dhclient.*		gen_context(system_u:object_r:dhcp_etc_t,s0)
@@ -36,6 +38,15 @@
 /etc/sysconfig/networking(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
 /etc/sysconfig/network-scripts(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
 ')
+/etc/udhcpc.d		-d	gen_context(system_u:object_r:dhcp_etc_t,s0)
+/etc/udhcpc.d/.+	--	gen_context(system_u:object_r:dhcp_etc_t,s0)
+
+/etc/iproute2(/.*)?	--	gen_context(system_u:object_r:net_conf_t,s0)
+
+#
+# /tmp
+#
+/tmp/hosts		--	gen_context(system_u:object_r:net_conf_t,s0)
 
 #
 # /usr
@@ -75,6 +86,10 @@
 /usr/sbin/mii-tool\.net-tools		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/pump			--	gen_context(system_u:object_r:dhcpc_exec_t,s0)
 /usr/sbin/tc			--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
+/usr/sbin/udhcpc		--	gen_context(system_u:object_r:dhcpc_exec_t,s0)
+
+/usr/share/udhcpc		-d	gen_context(system_u:object_r:dhcp_etc_t,s0)
+/usr/share/udhcpc/default\.script	--	gen_context(system_u:object_r:dhcp_etc_t,s0)
 
 #
 # /var
@@ -100,3 +115,11 @@
 /run/network(/.*)?	gen_context(system_u:object_r:net_conf_t,s0)
 /run/resolvconf/.* --	gen_context(system_u:object_r:net_conf_t,s0)
 ')
+
+/var/volatile/etc/resolv\.conf.*	--	gen_context(system_u:object_r:net_conf_t,s0)
+
+#
+# /config
+#
+/config/system/iproute2(/.*)?  --      gen_context(system_u:object_r:net_conf_t,s0)
+
--- a/policy/modules/system/sysnetwork.if
+++ b/policy/modules/system/sysnetwork.if
@@ -581,7 +581,8 @@
 	')
 
 	files_search_etc($1)
-	allow $1 net_conf_t:file manage_file_perms;
+	allow $1 net_conf_t:file read_file_perms;
+	allow $1 net_conf_t:lnk_file read_lnk_file_perms;
 
 	ifdef(`distro_debian',`
 		files_search_runtime($1)
@@ -957,6 +958,24 @@
 
 	files_search_var_lib($1)
 	allow $1 dhcp_state_t:dir search_dir_perms;
+')
+
+########################################
+## <summary>
+##	Dontaudit attemps to write to state dir.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`sysnet_dontaudit_write_dhcp_state',`
+	gen_require(`
+		type dhcp_state_t;
+	')
+
+	dontaudit $1 dhcp_state_t:dir write;
 ')
 
 ########################################
@@ -1107,3 +1126,31 @@
 
 	sysnet_read_config($1)
 ')
+
+########################################
+## <summary>
+##	Read dhcpc PID files (OpenXT).
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`sysnet_read_dhcpc_pid',`
+	sysnet_read_dhcpc_runtime_files($1)
+')
+
+########################################
+## <summary>
+##	Delete dhcpc PID files (OpenXT).
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`sysnet_delete_dhcpc_pid',`
+	sysnet_delete_dhcpc_runtime_files($1)
+')
--- a/policy/modules/system/sysnetwork.te
+++ b/policy/modules/system/sysnetwork.te
@@ -23,6 +23,7 @@
 # this is shared between dhcpc and dhcpd:
 type dhcp_state_t;
 files_type(dhcp_state_t)
+files_mountpoint(dhcp_state_t)
 
 type dhcpc_t;
 type dhcpc_exec_t;
@@ -51,7 +52,8 @@
 ')
 
 type net_conf_t;
-files_config_file(net_conf_t)
+files_type(net_conf_t)
+files_mountpoint(net_conf_t)
 
 ifdef(`distro_debian',`
 	init_daemon_runtime_file(net_conf_t, dir, "network")
@@ -116,6 +118,7 @@
 
 corecmd_exec_bin(dhcpc_t)
 corecmd_exec_shell(dhcpc_t)
+libs_exec_lib_files(dhcpc_t)
 
 corenet_all_recvfrom_netlabel(dhcpc_t)
 corenet_tcp_sendrecv_all_if(dhcpc_t)
@@ -132,9 +135,10 @@
 corenet_tcp_connect_all_ports(dhcpc_t)
 corenet_sendrecv_dhcpd_client_packets(dhcpc_t)
 corenet_sendrecv_all_server_packets(dhcpc_t)
-corenet_sendrecv_icmp_packets(dhcpc_t)
+corenet_udp_bind_all_ports(dhcpc_t)
 
 dev_read_sysfs(dhcpc_t)
+# for SSP:
 # for SSP:
 dev_read_rand(dhcpc_t)
 dev_read_urand(dhcpc_t)
@@ -144,6 +148,12 @@
 
 files_read_etc_files(dhcpc_t)
 files_read_etc_runtime_files(dhcpc_t)
+# to allow reading of /var/run/resolv.conf ... should consider a better place
+#   for resolv.conf with a more descriptive type
+files_read_all_runtime_files(dhcpc_t)
+# to allow reading of NetworkManager configuration from
+#   /var/run/nm-dhclient-brbridged.conf
+networkmanager_read_runtime_files(dhcpc_t)
 files_read_usr_files(dhcpc_t)
 files_search_home(dhcpc_t)
 files_search_var_lib(dhcpc_t)
@@ -226,11 +236,10 @@
 	modutils_run(dhcpc_t, dhcpc_roles)
 ')
 
+# run ping in the dhcpc_t domain without domtrans (busybox ip)
 # for the dhcp client to run ping to check IP addresses
-optional_policy(`
-	netutils_run_ping(dhcpc_t, dhcpc_roles)
-	netutils_run(dhcpc_t, dhcpc_roles)
-')
+netutils_run_ping(dhcpc_t, dhcpc_roles)
+netutils_run(dhcpc_t, dhcpc_roles)
 
 optional_policy(`
 	networkmanager_domtrans(dhcpc_t)
@@ -334,6 +343,7 @@
 kernel_read_system_state(ifconfig_t)
 kernel_read_network_state(ifconfig_t)
 kernel_request_load_module(ifconfig_t)
+kernel_load_module(ifconfig_t)
 kernel_search_network_sysctl(ifconfig_t)
 kernel_rw_net_sysctls(ifconfig_t)
 kernel_search_debugfs(ifconfig_t)
@@ -344,6 +354,9 @@
 dev_dontaudit_rw_cardmgr(ifconfig_t)
 # for IPSEC setup:
 dev_read_urand(ifconfig_t)
+
+# Early devtmpfs /dev/console access prior to udev relabel
+dev_dontaudit_rw_generic_chr_files(ifconfig_t)
 
 domain_use_interactive_fds(ifconfig_t)
 
@@ -374,6 +387,7 @@
 miscfiles_read_localization(ifconfig_t)
 
 seutil_use_runinit_fds(ifconfig_t)
+sysnet_read_config(ifconfig_t)
 
 sysnet_dontaudit_rw_dhcpc_udp_sockets(ifconfig_t)
 
