Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/roles/sysadm.te
+++ b/policy/modules/roles/sysadm.te
@@ -76,6 +76,8 @@
 # for systemd-analyze
 files_get_etc_unit_status(sysadm_t)
 
+storage_rw_blktap(sysadm_t)
+
 ifdef(`direct_sysadm_daemon',`
 	optional_policy(`
 		init_run_daemon(sysadm_t, sysadm_r)
@@ -1082,6 +1084,7 @@
 
 optional_policy(`
 	ssh_role_template(sysadm, sysadm_t, sysadm_application_exec_domain, sysadm_r)
+ssh_inherit_env(sysadm_t)
 ')
 
 optional_policy(`
