Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/system/xen.fc
+++ b/policy/modules/system/xen.fc
@@ -38,3 +38,7 @@
 /run/xenstored(/.*)?	gen_context(system_u:object_r:xenstored_runtime_t,s0)
 
 /xen(/.*)?	gen_context(system_u:object_r:xen_image_t,s0)
+/usr/bin/xenmgr         --      gen_context(system_u:object_r:xend_exec_t,s0)
+/var/lib/xenvm(/.*)?            gen_context(system_u:object_r:xend_var_lib_t,s0)
+/run/xen/qmp-libxl-.*	-s	gen_context(system_u:object_r:xend_runtime_t,s0)
+/config/system/dom0-networking-disabled         --              gen_context(system_u:object_r:xend_config_t,s0)
--- a/policy/modules/system/xen.if
+++ b/policy/modules/system/xen.if
@@ -206,3 +206,261 @@
 	files_search_runtime($1)
 	stream_connect_pattern($1, xenstored_runtime_t, xenstored_runtime_t, xm_t)
 ')
+
+########################################
+## <summary>
+##	Execute xm admin programs in the xm domain and
+## 	allow the specified role in the xm domain
+## </summary>
+## <param name="domain">
+##      <summary>
+##	Domain allowed access.
+##      </summary>
+## </param>
+## <param name="role">
+##      <summary>
+##	The role to allow the domain.
+##      </summary>
+## </param>
+#
+interface(`xen_run_xm', `
+	gen_require(`
+		type xm_t;
+	')
+
+	xen_domtrans_xm($1)
+	role $2 types xm_t;
+')
+
+########################################
+## <summary>
+##      Allow the specified domain to associate
+##      with xend's semaphores.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`xen_sem_use',`
+	gen_require(`
+		type xend_t;
+	')
+
+	allow $1 xend_t:sem { associate };
+')
+
+
+	allow $1 xend_t:sem { associate };
+')
+
+#####################################
+## <summary>
+##      Allow read and write access to xend semaphores.
+##      Added for dmsetup (lvm_t) run by xend.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`xend_rw_semaphores',`
+	gen_require(`
+		type xend_t;
+	')
+
+	allow $1 xend_t:sem rw_sem_perms;
+')
+
+########################################
+## <summary>
+##      Do not audit attempts to read and write
+##      Xen fifo files.  These are leaked file descriptors.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain to not audit.
+##      </summary>
+## </param>
+#
+interface(`xen_dontaudit_rw_unix_fifo_files',`
+	gen_require(`
+		type xend_t;
+	')
+
+	dontaudit $1 xend_t:fifo_file { read write };
+')
+
+
+	dontaudit $1 xend_t:fifo_file { read write };
+')
+
+#######################################
+## <summary>
+##      Exchange messages with xend over dbus.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      The type of the process chatting with xend over dbus.
+##      </summary>
+## </param>
+#
+interface(`xen_dbus_chat',`
+	gen_require(`
+		type xend_t;
+		class dbus send_msg;
+	')
+
+	allow $1 xend_t:dbus	send_msg;
+	allow xend_t $1:dbus	send_msg;
+')
+
+
+	dontaudit $1 xend_t:fifo_file { read write };
+')
+
+#######################################
+## <summary>
+##      Exchange messages with xend over dbus.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      The type of the process chatting with xend over dbus.
+##      </summary>
+## </param>
+#
+
+	allow $1 xend_t:dbus	send_msg;
+	allow xend_t $1:dbus	send_msg;
+')
+
+#######################################
+## <summary>
+##      Send messages to xend over dbus.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      The type of the process chatting with xend over dbus.
+##      </summary>
+## </param>
+#
+interface(`xen_dbus_send',`
+	gen_require(`
+		type xend_t;
+		class dbus send_msg;
+	')
+
+	allow $1 xend_t:dbus	send_msg;
+')
+
+########################################
+## <summary>
+##      Write to xen tmp files.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`xen_write_tmp',`
+	gen_require(`
+		type xend_tmp_t;
+	')
+
+	files_search_tmp($1)
+	allow $1 xend_tmp_t:file { write_file_perms };
+')
+
+########################################
+## <summary>
+##      Get attributes of files marked as xen config files
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`xen_geattr_config_files',`
+	gen_require(`
+		type xend_config_t;
+	')
+
+	getattr_files_pattern($1, xend_config_t, xend_config_t)
+')
+
+########################################
+## <summary>
+##	Do not audit attempts to read and write
+##	Xen unix domain stream sockets.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`xen_dontaudit_rw_unix_stream_sockets',`
+	gen_require(`
+		type xend_t;
+	')
+
+	dontaudit $1 xend_t:unix_stream_socket rw_stream_socket_perms;
+')
+
+########################################
+## <summary>
+##	Allow use of xend file descriptors.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`xen_use_fds',`
+	gen_require(`
+		type xend_t;
+	')
+
+	allow $1 xend_t:fd use;
+')
+
+########################################
+## <summary>
+##	Do not audit use of xend file descriptors.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`xen_dontaudit_use_fds',`
+	gen_require(`
+		type xend_t;
+	')
+
+	dontaudit $1 xend_t:fd use;
+')
+########################################
+## <summary>
+##	Execute a domain transition to run xend.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+interface(`xen_domtrans',`
+	gen_require(`
+		type xend_t, xend_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, xend_exec_t, xend_t)
+')
--- a/policy/modules/system/xen.te
+++ b/policy/modules/system/xen.te
@@ -81,6 +81,31 @@
 type xenctl_t;
 files_type(xenctl_t)
 
+type xend_t;
+type xend_exec_t;
+init_daemon_domain(xend_t, xend_exec_t)
+
+ifdef(`enable_mcs',`
+	init_ranged_daemon_domain(xend_t, xend_exec_t, s0 - mcs_systemhigh)
+')
+
+type xend_tmp_t;
+files_tmp_file(xend_tmp_t)
+files_mountpoint(xend_tmp_t)
+
+# OpenXT lock type (distinct from xen_lock_t used by xm)
+type xend_lock_t;
+files_lock_file(xend_lock_t)
+files_lock_filetrans(xend_t, xend_lock_t, file)
+
+# config for xenstored (OpenXT)
+type xenstored_etc_t;
+files_config_file(xenstored_etc_t)
+
+# XenClient /config files
+type xend_config_t;
+files_type(xend_config_t)
+
 type xenstored_t;
 type xenstored_exec_t;
 init_daemon_domain(xenstored_t, xenstored_exec_t)
@@ -146,6 +171,279 @@
 
 ########################################
 #
+# xend local policy
+#
+
+allow xend_t self:capability { dac_override ipc_lock net_admin net_raw setuid sys_admin sys_nice sys_rawio sys_resource sys_tty_config };
+dontaudit xend_t self:capability { sys_ptrace };
+allow xend_t self:process { setrlimit signal sigkill execmem setsched };
+dontaudit xend_t self:process ptrace;
+allow xend_t self:fifo_file rw_fifo_file_perms;
+allow xend_t self:unix_stream_socket { accept listen create_stream_socket_perms};
+allow xend_t self:tcp_socket { accept listen create_stream_socket_perms};
+allow xend_t self:unix_dgram_socket create_socket_perms;
+allow xend_t self:netlink_route_socket r_netlink_socket_perms;
+allow xend_t self:packet_socket create_socket_perms;
+allow xend_t self:tun_socket create_socket_perms;
+
+# XenClient /config files
+manage_dirs_pattern(xend_t, xend_config_t, xend_config_t)
+manage_files_pattern(xend_t, xend_config_t, xend_config_t)
+
+allow xend_t xen_image_t:dir list_dir_perms;
+manage_dirs_pattern(xend_t, xen_image_t, xen_image_t)
+manage_fifo_files_pattern(xend_t, xen_image_t, xen_image_t)
+manage_files_pattern(xend_t, xen_image_t, xen_image_t)
+read_lnk_files_pattern(xend_t, xen_image_t, xen_image_t)
+read_sock_files_pattern(xend_t, xen_image_t, xen_image_t)
+rw_chr_files_pattern(xend_t, xen_image_t, xen_image_t)
+rw_blk_files_pattern(xend_t, xen_image_t, xen_image_t)
+fs_hugetlbfs_filetrans(xend_t, xen_image_t, file)
+
+allow xend_t xenctl_t:fifo_file manage_fifo_file_perms;
+dev_filetrans(xend_t, xenctl_t, fifo_file)
+
+manage_files_pattern(xend_t, xend_tmp_t, xend_tmp_t)
+manage_dirs_pattern(xend_t, xend_tmp_t, xend_tmp_t)
+files_tmp_filetrans(xend_t, xend_tmp_t, { file dir })
+
+manage_dirs_pattern(xend_t, xend_runtime_t, xend_runtime_t)
+manage_files_pattern(xend_t, xend_runtime_t, xend_runtime_t)
+manage_sock_files_pattern(xend_t, xend_runtime_t, xend_runtime_t)
+manage_fifo_files_pattern(xend_t, xend_runtime_t, xend_runtime_t)
+files_runtime_filetrans(xend_t, xend_runtime_t, { file sock_file fifo_file dir })
+
+manage_files_pattern(xend_t, xend_var_log_t, xend_var_log_t)
+manage_dirs_pattern(xend_t, xend_var_log_t, xend_var_log_t)
+append_files_pattern(xend_t, xend_var_log_t, xend_var_log_t)
+create_files_pattern(xend_t, xend_var_log_t, xend_var_log_t)
+setattr_files_pattern(xend_t, xend_var_log_t, xend_var_log_t)
+manage_sock_files_pattern(xend_t, xend_var_log_t, xend_var_log_t)
+logging_log_filetrans(xend_t, xend_var_log_t, { sock_file file dir })
+
+# lock files
+manage_files_pattern(xend_t, xend_lock_t, xend_lock_t)
+
+manage_dirs_pattern(xend_t, xend_var_lib_t, xend_var_lib_t)
+manage_files_pattern(xend_t, xend_var_lib_t, xend_var_lib_t)
+manage_sock_files_pattern(xend_t, xend_var_lib_t, xend_var_lib_t)
+manage_fifo_files_pattern(xend_t, xend_var_lib_t, xend_var_lib_t)
+files_var_lib_filetrans(xend_t, xend_var_lib_t, { file dir })
+
+manage_files_pattern(xend_t, xenstored_runtime_t, xenstored_runtime_t)
+
+allow xend_t xenstored_var_lib_t:dir list_dir_perms;
+
+domtrans_pattern(xend_t, xenconsoled_exec_t, xenconsoled_t)
+domtrans_pattern(xend_t, xenstored_exec_t, xenstored_t)
+
+xen_stream_connect_xenstore(xend_t)
+
+# transition to qemu
+qemu_domtrans(xend_t)
+qemu_signal(xend_t)
+qemu_signull(xend_t)
+
+# transition to dmidecode
+dmidecode_domtrans(xend_t)
+
+kernel_read_kernel_sysctls(xend_t)
+kernel_read_system_state(xend_t)
+kernel_write_xen_state(xend_t)
+kernel_read_xen_state(xend_t)
+kernel_rw_net_sysctls(xend_t)
+kernel_read_network_state(xend_t)
+kernel_read_vm_sysctls(xend_t)
+
+corecmd_exec_bin(xend_t)
+corecmd_exec_shell(xend_t)
+
+corenet_all_recvfrom_netlabel(xend_t)
+corenet_tcp_sendrecv_generic_if(xend_t)
+corenet_tcp_sendrecv_generic_node(xend_t)
+corenet_tcp_bind_generic_node(xend_t)
+
+corenet_sendrecv_xen_server_packets(xend_t)
+corenet_tcp_bind_xen_port(xend_t)
+
+corenet_sendrecv_soundd_server_packets(xend_t)
+corenet_tcp_bind_soundd_port(xend_t)
+
+corenet_sendrecv_generic_server_packets(xend_t)
+corenet_tcp_bind_generic_port(xend_t)
+
+corenet_sendrecv_vnc_server_packets(xend_t)
+corenet_tcp_bind_vnc_port(xend_t)
+
+corenet_sendrecv_xserver_client_packets(xend_t)
+corenet_tcp_connect_xserver_port(xend_t)
+
+corenet_sendrecv_xen_client_packets(xend_t)
+corenet_tcp_connect_xen_port(xend_t)
+
+corenet_rw_tun_tap_dev(xend_t)
+
+dev_getattr_all_chr_files(xend_t)
+dev_read_urand(xend_t)
+dev_filetrans_xen(xend_t)
+dev_filetrans_blktap(xend_t, "blktap-2")
+dev_rw_sysfs(xend_t)
+dev_rw_xen(xend_t)
+dev_read_rand(xend_t)
+dev_search_xen(xend_t)
+dev_manage_xen(xend_t)
+dev_read_raw_memory(xend_t)
+
+domain_dontaudit_read_all_domains_state(xend_t)
+domain_dontaudit_ptrace_all_domains(xend_t)
+
+files_read_etc_files(xend_t)
+files_read_kernel_symbol_table(xend_t)
+files_read_kernel_img(xend_t)
+files_manage_etc_runtime_files(xend_t)
+files_etc_filetrans_etc_runtime(xend_t, file)
+files_read_usr_files(xend_t)
+files_read_default_symlinks(xend_t)
+files_search_mnt(xend_t)
+files_rw_etc_files(xend_t)
+#fs_getattr_all_f_exec_usr_files(xend_t)
+
+# accessing vmlinuz from service VM disks
+kernel_read_unlabeled_files(xend_t)
+
+fs_getattr_all_fs(xend_t)
+fs_list_auto_mountpoints(xend_t)
+fs_read_dos_files(xend_t)
+fs_read_removable_blk_files(xend_t)
+fs_manage_xenfs_dirs(xend_t)
+fs_manage_xenfs_files(xend_t)
+
+storage_read_scsi_generic(xend_t)
+# for lsscsi
+storage_getattr_fixed_disk_dev(xend_t)
+
+term_setattr_generic_ptys(xend_t)
+term_getattr_all_ptys(xend_t)
+term_setattr_all_ptys(xend_t)
+term_use_generic_ptys(xend_t)
+term_use_ptmx(xend_t)
+term_getattr_pty_fs(xend_t)
+
+storage_read_scsi_generic(xend_t)
+storage_raw_read_fixed_disk(xend_t)
+storage_raw_write_fixed_disk(xend_t)
+storage_raw_read_removable_device(xend_t)
+sysnet_domtrans_dhcpc(xend_t)
+sysnet_signal_dhcpc(xend_t)
+sysnet_domtrans_ifconfig(xend_t)
+sysnet_dns_name_resolve(xend_t)
+sysnet_delete_dhcpc_pid(xend_t)
+sysnet_read_dhcpc_pid(xend_t)
+sysnet_rw_dhcp_config(xend_t)
+
+# adding argo addresses?
+sysnet_write_config(xend_t)
+# restart ssh daemons
+ssh_initrc_domtrans(xend_t)
+
+seutil_read_config(xend_t)
+seutil_read_default_contexts(xend_t)
+
+# runlevel stuff
+init_rw_utmp(xend_t)
+init_telinit(xend_t)
+init_stream_connect_script(xend_t)
+
+locallogin_dontaudit_use_fds(xend_t)
+
+logging_send_syslog_msg(xend_t)
+lvm_domtrans(xend_t)
+
+miscfiles_read_localization(xend_t)
+miscfiles_read_hwdata(xend_t)
+mount_domtrans(xend_t)
+
+netutils_domtrans(xend_t)
+networkmanager_domtrans(xend_t)
+networkmanager_dbus_chat(xend_t)
+networkmanager_signal(xend_t)
+
+sysnet_domtrans_dhcpc(xend_t)
+sysnet_signal_dhcpc(xend_t)
+sysnet_domtrans_ifconfig(xend_t)
+sysnet_dns_name_resolve(xend_t)
+sysnet_delete_dhcpc_runtime_files(xend_t)
+sysnet_read_dhcpc_runtime_files(xend_t)
+sysnet_rw_dhcp_config(xend_t)
+
+userdom_dontaudit_search_user_home_dirs(xend_t)
+userdom_dontaudit_use_user_terminals(xend_t)
+
+xen_stream_connect_xenstore(xend_t)
+
+# act as a dbus client
+dbus_connect_system_bus(xend_t)
+dbus_system_bus_client(xend_t)
+
+# stubdom helpers
+optional_policy(`
+        atapi_helper_domtrans(xend_t)
+        atapi_helper_signal(xend_t)
+        audio_helper_domtrans(xend_t)
+        audio_helper_signal(xend_t)
+        varstored_watch_domtrans(xend_t)
+')
+
+optional_policy(`
+        tcsd_search_lib(xend_t)
+')
+
+tunable_policy(`xen_use_fusefs',`
+	fs_manage_fusefs_dirs(xend_t)
+	fs_manage_fusefs_files(xend_t)
+	fs_read_fusefs_symlinks(xend_t)
+')
+
+tunable_policy(`xen_use_nfs',`
+	fs_manage_nfs_dirs(xend_t)
+	fs_manage_nfs_files(xend_t)
+	fs_read_nfs_symlinks(xend_t)
+')
+
+tunable_policy(`xen_use_samba',`
+	fs_manage_cifs_dirs(xend_t)
+	fs_manage_cifs_files(xend_t)
+	fs_read_cifs_symlinks(xend_t)
+')
+
+optional_policy(`
+	brctl_domtrans(xend_t)
+')
+
+optional_policy(`
+	consoletype_exec(xend_t)
+')
+
+optional_policy(`
+	lvm_domtrans(xend_t)
+')
+
+optional_policy(`
+	mount_domtrans(xend_t)
+')
+
+optional_policy(`
+	netutils_domtrans(xend_t)
+')
+
+optional_policy(`
+	virt_search_images(xend_t)
+	virt_read_config(xend_t)
+')
+
+
+########################################
+#
 # Xen console local policy
 #
 
@@ -199,6 +497,9 @@
 #
 # Xen store local policy
 #
+
+allow xenstored_t xenstored_etc_t:file read_file_perms;
+files_search_etc(xenstored_t)
 
 allow xenstored_t self:capability { dac_override ipc_lock sys_resource };
 allow xenstored_t self:unix_stream_socket { accept listen };
