Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/system/selinuxutil.te
+++ b/policy/modules/system/selinuxutil.te
@@ -124,6 +124,9 @@
 type setfiles_exec_t alias restorecon_exec_t;
 init_system_domain(setfiles_t, setfiles_exec_t)
 domain_obj_id_change_exemption(setfiles_t)
+term_read_console(setfiles_t)
+dev_read_urand(setfiles_t)
+dev_rw_generic_chr_files(setfiles_t)
 
 ########################################
 #
@@ -181,6 +184,7 @@
 
 dev_read_urand(load_policy_t)
 
+dev_dontaudit_search_sysfs(load_policy_t)
 domain_use_interactive_fds(load_policy_t)
 
 # for mcs.conf
@@ -228,6 +232,17 @@
 	unconfined_write_inherited_pipes(load_policy_t)
         # leaked file descriptors
         unconfined_dontaudit_read_pipes(load_policy_t)
+')
+
+# XC init leaks fds
+init_dontaudit_use_fds(load_policy_t)
+
+# leaked by opkg
+ifdef(`distro_debian',`
+	optional_policy(`
+		dpkg_dontaudit_manage_db(load_policy_t)
+		dpkg_dontaudit_lock_db(load_policy_t)
+	')
 ')
 
 ########################################
@@ -295,9 +310,10 @@
 auth_run_chk_passwd(newrole_t, newrole_roles)
 auth_run_upd_passwd(newrole_t, newrole_roles)
 auth_rw_faillog(newrole_t)
-auth_search_faillog(newrole_t)
+auth_read_shadow(newrole_t)
 
 # Write to utmp.
+init_rw_utmp(newrole_t)
 init_rw_utmp(newrole_t)
 init_use_fds(newrole_t)
 
@@ -550,6 +566,7 @@
 corecmd_exec_shell(semanage_t)
 
 dev_read_urand(semanage_t)
+dev_dontaudit_search_sysfs(semanage_t)
 
 domain_use_interactive_fds(semanage_t)
 
@@ -666,6 +683,14 @@
 # to read bin_t symlinks
 corecmd_search_bin(setfiles_t)
 
+# leaked by opkg
+ifdef(`distro_debian',`
+	optional_policy(`
+		dpkg_dontaudit_manage_db(setfiles_t)
+		dpkg_dontaudit_lock_db(setfiles_t)
+	')
+')
+
 domain_use_interactive_fds(setfiles_t)
 domain_dontaudit_search_all_domains_state(setfiles_t)
 
