Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/system/authlogin.fc
+++ b/policy/modules/system/authlogin.fc
@@ -60,6 +60,7 @@
 /var/log/lastlog	--	gen_context(system_u:object_r:lastlog_t,s0)
 /var/log/tallylog	--	gen_context(system_u:object_r:faillog_t,s0)
 /var/log/wtmp.*		--	gen_context(system_u:object_r:wtmp_t,s0)
+/var/volatile/log/wtmp.*	--	gen_context(system_u:object_r:wtmp_t,s0)
 
 /run/console(/.*)?	 	gen_context(system_u:object_r:pam_var_console_t,s0)
 /run/faillock(/.*)?		gen_context(system_u:object_r:faillog_t,s0)
@@ -74,3 +75,9 @@
 /run/sudo/ts/%{USERNAME}	gen_context(system_u:object_r:pam_runtime_t,s0)
 /var/(db|adm)/sudo(/.*)?	gen_context(system_u:object_r:pam_runtime_t,s0)
 /var/lib/sudo(/.*)?	gen_context(system_u:object_r:pam_runtime_t,s0)
+
+/config/etc/\.pwd\.lock	--	gen_context(system_u:object_r:shadow_t,s0)
+/config/etc/group\.lock	--	gen_context(system_u:object_r:shadow_t,s0)
+/config/etc/gshadow.*	--	gen_context(system_u:object_r:shadow_t,s0)
+/config/etc/passwd\.lock	--	gen_context(system_u:object_r:shadow_t,s0)
+/config/etc/shadow.*	--	gen_context(system_u:object_r:shadow_t,s0)
--- a/policy/modules/system/authlogin.if
+++ b/policy/modules/system/authlogin.if
@@ -1113,6 +1113,24 @@
 	delete_files_pattern($1, lastlog_t, lastlog_t)
 ')
 
+#######################################
+## <summary>
+##	Set attributes on PAM PID directories.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`auth_setattr_pam_pid_dirs',`
+	gen_require(`
+		type pam_var_run_t;
+	')
+
+	allow $1 pam_var_run_t:dir setattr;
+')
+
 ########################################
 ## <summary>
 ##     Manage the last logins log.
@@ -1659,6 +1677,7 @@
 		type wtmp_t;
 	')
 
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
 	allow $1 wtmp_t:file setattr;
 	logging_search_logs($1)
 ')
@@ -1681,6 +1700,7 @@
 
 	logging_search_logs($1)
 	allow $1 wtmp_t:file read_file_perms;
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
 ')
 
 ########################################
@@ -1699,7 +1719,7 @@
 	gen_require(`
 		type wtmp_t;
 	')
-
+	dontaudit $1 wtmp_t:lnk_file read_lnk_file_perms;
 	dontaudit $1 wtmp_t:file read_file_perms;
 ')
 
@@ -1719,6 +1739,7 @@
 		type wtmp_t;
 	')
 
+	dontaudit $1 wtmp_t:lnk_file read_lnk_file_perms;
 	dontaudit $1 wtmp_t:file write;
 ')
 
@@ -1737,6 +1758,7 @@
 		type wtmp_t;
 	')
 
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
 	allow $1 wtmp_t:file append_file_perms;
 	logging_search_logs($1)
 ')
@@ -1756,7 +1778,8 @@
 		type wtmp_t;
 	')
 
-	allow $1 wtmp_t:file { lock write_file_perms };
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
+	allow $1 wtmp_t:file { write_file_perms lock };
 ')
 
 ########################################
@@ -1774,6 +1797,7 @@
 		type wtmp_t;
 	')
 
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
 	allow $1 wtmp_t:file rw_file_perms;
 	logging_search_logs($1)
 ')
@@ -1794,6 +1818,7 @@
 		type wtmp_t;
 	')
 
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
 	logging_log_filetrans($1, wtmp_t, file)
 ')
 
@@ -1813,6 +1838,7 @@
 		type wtmp_t;
 	')
 
+	allow $1 wtmp_t:lnk_file read_lnk_file_perms;
 	logging_rw_generic_log_dirs($1)
 	allow $1 wtmp_t:file manage_file_perms;
 ')
