Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/kernel/files.fc
+++ b/policy/modules/kernel/files.fc
@@ -51,6 +51,8 @@
 #
 /etc			-d	gen_context(system_u:object_r:etc_t,s0)
 /etc/.*				gen_context(system_u:object_r:etc_t,s0)
+/var/volatile/etc	-d	gen_context(system_u:object_r:etc_t,s0)
+/var/volatile/etc/.*		gen_context(system_u:object_r:etc_t,s0)
 /etc/\.fstab\.hal\..+	--	gen_context(system_u:object_r:etc_runtime_t,s0)
 /etc/blkid(/.*)?		gen_context(system_u:object_r:etc_runtime_t,s0)
 /etc/cmtab		--	gen_context(system_u:object_r:etc_runtime_t,s0)
@@ -71,6 +73,8 @@
 /etc/ipsec\.d/examples(/.*)?	gen_context(system_u:object_r:etc_t,s0)
 
 /etc/network/ifstate	--	gen_context(system_u:object_r:etc_runtime_t,s0)
+/var/volatile/etc/ifstate	--	gen_context(system_u:object_r:etc_runtime_t,s0)
+/run/ifstate		--	gen_context(system_u:object_r:etc_runtime_t,s0)
 
 /etc/ptal/ptal-printd-like -- 	gen_context(system_u:object_r:etc_runtime_t,s0)
 
@@ -88,6 +92,8 @@
 /etc/defkeymap\.map	--	gen_context(system_u:object_r:etc_runtime_t,s0)
 /etc/rc\.d/init\.d/\.depend.* -- gen_context(system_u:object_r:etc_runtime_t,s0)
 ')
+
+/var/volatile/system-connections(/.*)?	gen_context(system_u:object_r:etc_t,s0)
 
 #
 # HOME_ROOT
@@ -275,3 +281,16 @@
 /var/tmp/systemd-private-[^/]+/tmp	-d	gen_context(system_u:object_r:tmp_t,s0-mls_systemhigh)
 /var/tmp/systemd-private-[^/]+/tmp/.*	<<none>>
 /var/tmp/vi\.recover	-d	gen_context(system_u:object_r:tmp_t,s0)
+
+/var/volatile/lock(/.*)?		gen_context(system_u:object_r:var_lock_t,s0)
+/var/volatile/tmp		-d	gen_context(system_u:object_r:tmp_t,s0)
+/var/volatile/tmp/.*			<<none>>
+/var/volatile/tmp/lost\+found	-d	gen_context(system_u:object_r:lost_found_t,mls_systemhigh)
+/var/volatile/tmp/lost\+found/.*	<<none>>
+/var/volatile/tmp/vi\.recover	-d	gen_context(system_u:object_r:tmp_t,s0)
+/var/volatile/run		-d	gen_context(system_u:object_r:var_run_t,s0-mls_systemhigh)
+/var/volatile/run/.*			gen_context(system_u:object_r:var_run_t,s0)
+/var/volatile/run/.*\.*pid		<<none>>
+
+/var/cores		-d	gen_context(system_u:object_r:var_core_t,s0)
+/var/cores/*		--	gen_context(system_u:object_r:var_core_t,s0)
--- a/policy/modules/kernel/files.if
+++ b/policy/modules/kernel/files.if
@@ -1352,6 +1352,7 @@
 	')
 
 	allow $1 file_type:dir list_dir_perms;
+	allow $1 file_type:lnk_file read_lnk_file_perms;
 ')
 
 ########################################
@@ -3097,6 +3098,25 @@
 	')
 
 	allow $1 default_t:lnk_file read_lnk_file_perms;
+')
+
+########################################
+## <summary>
+##	Do not audit attempts to get attributes of symbolic
+##	links with the default file type.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_getattr_default_symlinks',`
+	gen_require(`
+		type default_t;
+	')
+
+	dontaudit $1 default_t:lnk_file getattr_lnk_file_perms;
 ')
 
 ########################################
@@ -3368,6 +3388,7 @@
 
 	allow $1 etc_t:dir list_dir_perms;
 	read_files_pattern($1, etc_t, etc_t)
+	allow $1 etc_t:file map;
 	read_lnk_files_pattern($1, etc_t, etc_t)
 ')
 
@@ -3730,6 +3751,36 @@
 
 	allow $1 etc_runtime_t:file manage_file_perms;
 	filetrans_pattern($1, root_t, etc_runtime_t, file, $2)
+')
+
+########################################
+## <summary>
+##	Create a runtime configuration file.
+## </summary>
+## <desc>
+##	<p>
+##	Create a runtime configuration file, such as /run/ifstate.
+##	</p>
+## </desc>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="name" optional="true">
+##	<summary>
+##	The name of the object being created.
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`files_create_etc_runtime',`
+	gen_require(`
+		type var_run_t, etc_runtime_t;
+	')
+
+	allow $1 etc_runtime_t:file manage_file_perms;
+	filetrans_pattern($1, var_run_t, etc_runtime_t, file, $2)
 ')
 
 ########################################
@@ -4542,6 +4593,24 @@
 
 ########################################
 ## <summary>
+##	Do not audit attempts to getattrs of /mnt directory.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_getattr_mnt_dirs',`
+	gen_require(`
+		type mnt_t;
+	')
+
+	dontaudit $1 mnt_t:dir getattr_dir_perms;
+')
+
+########################################
+## <summary>
 ##	Create, read, write, and delete symbolic links in /mnt.
 ## </summary>
 ## <param name="domain">
@@ -4950,6 +5019,7 @@
 		type tmp_t;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	allow $1 tmp_t:dir getattr;
 ')
 
@@ -4969,6 +5039,7 @@
 		type tmp_t;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	dontaudit $1 tmp_t:dir getattr;
 ')
 
@@ -5006,6 +5077,7 @@
 		type tmp_t;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	dontaudit $1 tmp_t:dir search_dir_perms;
 ')
 
@@ -5043,6 +5115,7 @@
 		type tmp_t;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	dontaudit $1 tmp_t:dir list_dir_perms;
 ')
 
@@ -5155,6 +5228,7 @@
 		type tmp_t;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	read_lnk_files_pattern($1, tmp_t, tmp_t)
 ')
 
@@ -5210,6 +5284,7 @@
 		attribute tmpfile;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	allow $1 tmpfile:dir { search_dir_perms setattr };
 ')
 
@@ -5228,6 +5303,7 @@
 		attribute tmpfile;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	allow $1 tmpfile:dir list_dir_perms;
 ')
 
@@ -5309,6 +5385,7 @@
 		type var_t;
 	')
 
+	allow $1 var_lock_t:lnk_file read_lnk_file_perms;
 	allow $1 var_t:dir search_dir_perms;
 	relabel_files_pattern($1, tmpfile, tmpfile)
 ')
@@ -5400,6 +5477,7 @@
 		attribute tmpfile;
 	')
 
+	allow $1 tmp_t:lnk_file read_lnk_file_perms;
 	allow $1 tmpfile:dir list_dir_perms;
 	delete_dirs_pattern($1, tmpfile, tmpfile)
 	delete_files_pattern($1, tmpfile, tmpfile)
@@ -6013,15 +6091,34 @@
 #
 interface(`files_search_var',`
 	gen_require(`
+		type var_t, var_lock_t;
+	')
+
+	allow $1 var_lock_t:lnk_file read_lnk_file_perms;
+	allow $1 var_t:dir search_dir_perms;
+')
+
+########################################
+## <summary>
+##	Do not audit attempts to write to /var.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_write_var_dirs',`
+	gen_require(`
 		type var_t;
 	')
 
-	allow $1 var_t:dir search_dir_perms;
-')
-
-########################################
-## <summary>
-##	Do not audit attempts to write to /var.
+	dontaudit $1 var_t:dir write;
+')
+
+########################################
+## <summary>
+##	Do not audit attempts to write to /var/lock.
 ## </summary>
 ## <param name="domain">
 ##	<summary>
@@ -6029,12 +6126,12 @@
 ##	</summary>
 ## </param>
 #
-interface(`files_dontaudit_write_var_dirs',`
-	gen_require(`
-		type var_t;
-	')
-
-	dontaudit $1 var_t:dir write;
+interface(`files_dontaudit_write_lock_dirs',`
+	gen_require(`
+		type var_lock_t;
+	')
+
+	dontaudit $1 var_lock_t:dir write;
 ')
 
 ########################################
@@ -6499,9 +6596,10 @@
 #
 interface(`files_var_lib_filetrans',`
 	gen_require(`
-		type var_t, var_lib_t;
-	')
-
+		type var_t, var_lib_t, var_run_t;
+	')
+
+	allow $1 var_run_t:lnk_file read_lnk_file_perms;
 	allow $1 var_t:dir search_dir_perms;
 	filetrans_pattern($1, var_lib_t, $2, $3, $4)
 ')
@@ -6596,9 +6694,10 @@
 #
 interface(`files_manage_mounttab',`
 	gen_require(`
-		type var_t, var_lib_t;
-	')
-
+		type var_t, var_lib_t, var_run_t;
+	')
+
+	allow $1 var_run_t:lnk_file read_lnk_file_perms;
 	allow $1 var_t:dir search_dir_perms;
 	manage_files_pattern($1, var_lib_t, var_lib_t)
 ')
@@ -6971,6 +7070,69 @@
 
 ########################################
 ## <summary>
+##     Create an object in the locks directory, with a private
+##     type using both a type and a range transition.
+## </summary>
+## <param name="domain">
+##     <summary>
+##     Domain allowed access.
+##     </summary>
+## </param>
+## <param name="private type">
+##     <summary>
+##     The type of the object to be created.
+##     </summary>
+## </param>
+## <param name="object">
+##     <summary>
+##     The object class of the object being created.
+##     </summary>
+## </param>
+## <param name="range">
+##     <summary>
+##     Range for the object.
+##     </summary>
+## </param>
+#
+interface(`files_lock_ranged_filetrans',`
+       gen_require(`
+               type var_t, var_lock_t;
+       ')
+
+	allow $1 var_t:dir search_dir_perms;
+	allow $1 var_lock_t:lnk_file read_lnk_file_perms;
+	filetrans_pattern($1, var_lock_t, $2, $3)
+
+	ifdef(`enable_mcs',`
+		range_transition $1 var_lock_t:$3 $4;
+	')
+
+	ifdef(`enable_mls',`
+		range_transition $1 var_lock_t:$3 $4;
+	')
+')
+
+########################################
+## <summary>
+##	Create core dumps
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_create_core_dump',`
+	gen_require(`
+		type var_core_t;
+	')
+
+	allow $1 var_core_t:dir add_entry_dir_perms;
+	allow $1 var_core_t:file { create_file_perms rw_file_perms };
+')
+
+########################################
+## <summary>
 ##	Do not audit attempts to get the attributes
 ##	of the /var/run directory.
 ## </summary>
@@ -7009,6 +7171,25 @@
 
 ########################################
 ## <summary>
+##     Do not audit attempts to write to
+##     the /var/run directory.
+## </summary>
+## <param name="domain">
+##     <summary>
+##     Domain to not audit.
+##     </summary>
+## </param>
+#
+interface(`files_dontaudit_write_pid_dirs',`
+	gen_require(`
+		type var_run_t;
+	')
+
+	dontaudit $1 var_run_t:dir write;
+')
+
+########################################
+## <summary>
 ##	Set the attributes of the /var/run directory.
 ## </summary>
 ## <param name="domain">
@@ -7044,6 +7225,7 @@
 
 	allow $1 var_run_t:lnk_file read_lnk_file_perms;
 	search_dirs_pattern($1, var_t, var_run_t)
+	read_lnk_files_pattern($1, var_t, var_run_t)
 ')
 
 ########################################
@@ -7339,9 +7521,46 @@
 interface(`files_manage_all_runtime_dirs',`
 	gen_require(`
 		attribute pidfile;
+		type var_run_t;
 	')
 
 	manage_dirs_pattern($1, pidfile, pidfile)
+')
+
+########################################
+## <summary>
+##	Set attributes on all runtime data dirs.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_setattr_all_pid_dirs',`
+	gen_require(`
+		attribute pidfile;
+	')
+
+	allow $1 pidfile:dir setattr;
+')
+
+########################################
+## <summary>
+##	Set attributes on all runtime pid files.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_setattr_all_pids',`
+	gen_require(`
+		attribute pidfile;
+	')
+
+	allow $1 pidfile:file setattr;
 ')
 
 ########################################
@@ -8031,3 +8250,21 @@
 
 	typeattribute $1 files_unconfined_type;
 ')
+
+########################################
+## <summary>
+##      Unconfined access to files.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`files_unconfined_readonly',`
+	gen_require(`
+		attribute files_unconfined_readonly_type;
+	')
+
+	typeattribute $1 files_unconfined_readonly_type;
+')
--- a/policy/modules/kernel/files.te
+++ b/policy/modules/kernel/files.te
@@ -6,6 +6,7 @@
 #
 
 attribute file_type;
+attribute files_unconfined_readonly_type;
 attribute files_unconfined_type;
 attribute lockfile;
 attribute mountpoint;
@@ -61,6 +62,9 @@
 #
 type etc_t, configfile;
 files_type(etc_t)
+# XC: this is needed to mount a tmpfs on /etc/NetworkManager/system-connections
+files_mountpoint(etc_t)
+
 
 optional_policy(`
 	# for systemd ProtectSystem
@@ -185,6 +189,13 @@
 optional_policy(`
 	systemd_tmpfilesd_managed(var_run_t)
 ')
+
+#
+# var_core_t is the type of /var/cores, used
+# for core dumps.
+#
+type var_core_t;
+files_mountpoint(var_core_t)
 
 #
 # var_spool_t is the type of /var/spool
@@ -241,3 +252,14 @@
 tunable_policy(`allow_execmod',`
 	allow files_unconfined_type file_type:file execmod;
 ')
+
+########################################
+#
+# Unconfined readonly access to this module
+#
+
+allow files_unconfined_readonly_type file_type:{ file sock_file fifo_file blk_file } {  getattr open read lock ioctl };
+allow files_unconfined_readonly_type file_type: lnk_file { read_lnk_file_perms getattr_lnk_file_perms };
+allow files_unconfined_readonly_type file_type: sock_file write_sock_file_perms ;
+allow files_unconfined_readonly_type file_type: chr_file read_chr_file_perms ;
+allow files_unconfined_readonly_type file_type: dir { search_dir_perms getattr_dir_perms list_dir_perms };
