Upstream-Status: Inappropriate [OpenXT]

--- a/policy/modules/roles/sysadm.te
+++ b/policy/modules/roles/sysadm.te
@@ -1342,6 +1342,7 @@
 	viptables_run(sysadm_t, sysadm_r)
 	xentop_run(sysadm_t, sysadm_r)
 	xec_run(sysadm_t, sysadm_r)
+	xen_run_xm(sysadm_t, sysadm_r)
 ')
 
 ifndef(`distro_redhat',`
--- a/policy/modules/system/xen.te
+++ b/policy/modules/system/xen.te
@@ -767,3 +767,5 @@
 	fs_manage_xenfs_dirs(xm_ssh_t)
 	fs_manage_xenfs_files(xm_ssh_t)
 ')
+seutil_use_newrole_fds(xm_t)
+
